File: 3. Scripts/javascript/singup_page.js
class SignupPage {
  // CSS selectors
  EMAIL_FIELD = { id: 'email' };
  PASSWORD_FIELD = { id: 'password_first' };
  PASSWORD2_FIELD = { id: 'password_second' };
  TOS_BOX = { id: 'tos' };
  EASYSALECONSENT_BOX = { id: 'easySaleConsent' };
  EASYBUYCONSENT_BOX = { id: 'easySaleConsent' };
  REG_BUTTON = { name: 'login' };

  constructor(driver) {
    this.driver = driver;
  }

  async enterEmailAddress(email) {
    const emailField = await this.driver.findElement(this.EMAIL_FIELD);
    await emailField.sendKeys(email);
  }

  async enterPassword(password) {
    const passwordField = await this.driver.findElement(this.PASSWORD_FIELD);
    await passwordField.sendKeys(password);
  }

  async enterPassword2(password2) {
    const password2Field = await this.driver.findElement(this.PASSWORD2_FIELD);
    await password2Field.sendKeys(password2);
  }

  async clickRules() {
    const tosBox = await this.driver.findElement(this.TOS_BOX);
    await tosBox.click();
  }

  async clickEasySale() {
    const easySaleConsentBox = await this.driver.findElement(this.EASYSALECONSENT_BOX);
    await easySaleConsentBox.click();
  }

  async clickEasyBuy() {
    const easyBuyConsentBox = await this.driver.findElement(this.EASYBUYCONSENT_BOX);
    await easyBuyConsentBox.click();
  }

  async clickLogin() {
    const regButton = await this.driver.findElement(this.REG_BUTTON);
    await regButton.click();
  }
}

File: 3. Scripts/ruby/singup_page.rb
class SignupPage


  #css selectors
  EMAIL_FIELD = {id: 'email'}
  PASSWORD_FIELD = {id: 'password_first'}
  PASSWORD2_FIELD = {id: 'password_second'}
  TOS_BOX = {id: 'tos'}
  EASYSALECONSENT_BOX = {id: 'easySaleConsent'}
  EASYBUYCONSENT_BOX = {id: 'easySaleConsent'}
  REG_BUTTON = {name: 'login'}


  attr_reader :driver

  def initialize(driver)
    @driver = driver
  end

  def  enter_email_address(email)
    email_field = @driver.find_element(EMAIL_FIELD)
    email_field.send_keys(email)

  end

  def  enter_password(password)
    password_field = @driver.find_element(PASSWORD_FIELD)
    password_field.send_keys(password)

  end

  def  enter_password2(password2)
    password2_field = @driver.find_element(PASSWORD2_FIELD)
    password2_field.send_keys(password2)

  end

  def click_rules()
    tos_box = @driver.find_element(TOS_BOX)
    tos_box.click
  end

  def click_easysale()
    easySaleConsent_box = driver.find_element(EASYSALECONSENT_BOX)
    easySaleConsent_box.click
  end

  def click_easybuy()
    easyBuyConsent_box = driver.find_element(EASYBUYCONSENT_BOX)
    easyBuyConsent_box.click
  end

  def click_login()
    reg_button = driver.find_element(REG_BUTTON)
    reg_button.click
  end


  
end